### ISSUE TITLE
[GraphQL] Relations need `item_query` and `collection_query` to be set

### ISSUE THREAD

**Issue by lukasluecke** [api-platform/core#3136]
Right now setting `item_query` and `collection_query` on a Resource will add these queries both to the root `Query` type, and also as fields to all other Resources that have a relation to this one.

I would like to make certain queries only available through relations, without "polluting" the root Query.

Disabling / removing `item_query` and `collection_query` seem to also remove the respective relation fields.

**Comment by alanpoulain** [comment]
The relations are resources (they use the `@ApiResource` annotation), right? It seems logical to me that the root queries are created. If the user doesn't want them, it can disable the operations.
It's the same as REST.

**Comment by lukasluecke** [comment]
If I disable the operations, then the relations are also not available - that's the issue 🙈 Sorry if my description was not clear enough.

**Comment by alanpoulain** [comment]
It's a bug then since it should be.

**Comment by lukasluecke** [comment]
Okay, so the relations should always be created? And the `item_query` and `collection_query` should only affect the root queries?

I'll try to hunt down the issue then 🙂

**Comment by lukasluecke** [comment]
@alanpoulain The problem seems to be here: https://github.com/api-platform/core/blob/e468d98d260e0cf49d63d088fd5678ecab1b8912/src/GraphQl/Type/TypeConverter.php#L106-L110

An empty `graphql` array should mean that the Resource is still available to be exposed, just not directly accessible. What do you think?

**Comment by alanpoulain** [comment]
It can be tried like this, yes.

**Comment by lukasluecke** [comment]
How about changing the condition to the following? Would that be the correct behaviour? I'm not sure which cases the previous fallback was supposed to handle 🙈 

```php
if (null === $resourceMetadata->getGraphql()) {
    return null;
}
```

**Comment by alanpoulain** [comment]
Will be fixed in 2.5.6.
